packages/nouns-contracts/test/foundry/StreamEscrow.t.sol

```diff
@@ -0,0 +1,718 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+import { Test } from 'forge-std/Test.sol';
+import { StreamEscrow } from '../../contracts/StreamEscrow.sol';
+import { ERC721Mock } from './helpers/ERC721Mock.sol';
+import 'forge-std/console.sol';
+
+abstract contract BaseStreamEscrowTest is Test {
+    StreamEscrow escrow;
+    address treasury = makeAddr('treasury');
+    address ethRecipient = makeAddr('ethRecipient');
+    address nounsRecipient = makeAddr('nounsRecipient');
+    address streamCreator = makeAddr('streamCreator');
+    ERC721Mock nounsToken = new ERC721Mock();
+    address user = makeAddr('user');
+
+    function setUp() public virtual {
+        escrow = new StreamEscrow(treasury, ethRecipient, nounsRecipient, address(nounsToken), streamCreator);
+
+        nounsToken.mint(streamCreator, 1);
+        vm.deal(streamCreator, 1000 ether);
+    }
+
+    function forwardOneDay() internal {
+        vm.warp(block.timestamp + 24 hours);
+        escrow.forwardAll();
+    }
+}
+
+contract CreateStreamPermissionsTest is BaseStreamEscrowTest {
+    function test_createStream_failsIfNotWhitelisted() public {
+        vm.expectRevert('not allowed');
+        escrow.createStream(1, 1000);
+    }
+
+    function test_nounOwner_failsIfNotWhitelisted() public {
```

Comment:
nit: change function name to start the same as others `test_createStream`

Reply:
done 7689e3bed873db0060f8ad2f6dfbbeb937999f4c